src/utils/Types.tsx

```diff
@@ -31,96 +29,99 @@ export interface row {
   index: number;
 
   [key: string]: RowProp;
-}
+};
 
-export interface combinedRow {
+export type CombinedRow = {
   index: number;
   averageCoordinates: Point;
-  rows: row[];
+  rows: Row[];
   isMarkerVisible: Visibility;
 
   [key: string]: CombinedProp;
-}
+};
 
-export interface position {
+export type Position = {
   coordinates: Point;
   zoom: number;
-}
+};
+
+export type Action = {
+  type: string;
+  value: StateProp;
+};
+export type State = {
+  rows: Row[];
+  allCombinedRows: CombinedRow[];
+  tooltipContent: string;
+  currentCombinedRows: CombinedRow[];
+  mousePosition: Position;
+  searchBarContent: string;
+  useMarkerVisibility: boolean;
+  useSearchBar: boolean;
+  inFullMode: boolean;
+  currentRow: Row;
+};
+export type Dispatch = React.Dispatch<Action>;
+export type StateManager = [State, Dispatch];
+
+export type MarkerDetail = {
+  key: string;
+  header: string;
+  getRowPropContent: (value: any) => string;
+};
+
+export type Limits = {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+};
 
 export interface SideBarProps {
-  stateManager: stateManager;
+  stateManager: StateManager;
   width: string;
 }
 
 export interface FilterOptionsProps {
-  stateManager: stateManager;
+  stateManager: StateManager;
 }
 
 export interface OrphanTableRowsProps {
   keys: string[];
-  stateManager: stateManager;
-  givenCombinedRow: combinedRow;
-  givenIndex: number;
```

Comment:
givenIndex is no longer used since I used CSS to handle the odd/even row styling